src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentTracker.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public abstract class RazorDocumentTracker
+    {
+        public abstract event EventHandler ContextChanged;
+
+        public abstract bool IsSupportedDocument { get; }
```

Comment:
What's this guy supposed to represent? Are we expecting document trackers to be produced for non-valid projects?